hw/drivers/uart/max3107/max3107_0/syscfg.yml

```diff
@@ -0,0 +1,125 @@
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+syscfg.defs:
+    MAX3107_0_NAME:
+        description: >
+            MAX3107 device name.
+        value: '"max3107_0"'
+
+    MAX3107_0_UART_NAME:
+        description: >
+            MAX3107 UART device name.
+        value: '"uart4"'
+
+    MAX3107_0_SPI_BAUDRATE:
+        description: >
+            MAX3107 SPI clock speed in kHz.
+        value: 4000
+
+    MAX3107_0_UART_BAUDRATE:
+        description: >
+            MAX3107 UART clock speed in bits/second.
+        value: 115200
+
+    MAX3107_0_UART_DATA_BITS:
+        description: >
+            MAX3107 number of data bits.
+        range: 5,6,7,8
+        value: 8
+
+    MAX3107_0_UART_PARITY:
+        description: >
+            MAX3107 parity 0 - no parity, 1 - odd, 2 - even.
+        range: 0,1,2
+        value: 0
+
+    MAX3107_0_UART_STOP_BITS:
+        description: >
+            MAX3107 UART stop bits 1 or 2.
+        range: 1,2
+        value: 1
+
+    MAX3107_0_UART_FLOW_CONTROL:
+        description: >
+            MAX3107 UART flow control 0 - none, 1 - RTS CTS.
+        range: 0,1
+        value: 0
+
+    MAX3107_0_UART_RX_FIFO_LEVEL:
+        description: >
+            RX FIFO level at which FIFO trigger is activated.
+            (Value will be rounded down to be multiply of 8).
+        range: 0..120
+        value: 24
+
+    MAX3107_0_UART_TX_FIFO_LEVEL:
+        description: >
+            TX FIFO level at which FIFO trigger is activated.
+            (Value will be rounded down to be multiply of 8).
+        range: 0..120
+        value: 64
+
+    MAX3107_0_CS_PIN:
+        description: >
+            Chip select pin for MAX3107.
+        value: -1
+
+    MAX3107_0_SPI_BUS:
+        description: >
+            SPI interface bus name for MAX3107.
+        value : '"spi0"'
+
+    MAX3107_0_SPI_NUM:
+        description: >
+            SPI interface used for MAX3107.
+        value: 0
+
+    MAX3107_0_OSC_FREQ:
+        description: >
+            External clock/oscillator frequency.
+        range: 500000..35000000
+        value: 0
+
+    MAX3107_0_CRYSTAL_EN:
+        description: >
+            Set 0 when external clock is connected to XIN.
+            Set 1 when external crystal oscillator is connected to XIN, XOUT.
+        value: 0
+        range: 0,1
+
+    MAX3107_0_DISABLE_PLL:
+        description: >
+            MAX3170 PLL is not to be used.
+            Settings this to 1 limits available baud rates.
+        value: 0
+
+    MAX3107_0_IRQ_PIN:
+        description: >
+            MAX3107 interrupt pin.
+        value: -1
+
+    MAX3107_0_LDOEN_PIN:
+        description: >
+            MAX3107 LDEON pin.
+        value: -1
+
```

Comment:
All above instances should not be _0_ I think, it's best to keep only one instance.

Reply:
I disagree